modules/rest/src/main/java/org/apache/ignite/internal/rest/RestModule.java

```diff
@@ -232,34 +244,36 @@ private void handleRepresentByPath(
      * @param res          Rest response.
      * @param presentation Configuration presentation.
      */
-    private void handleUpdate(
+    private static CompletableFuture<RestApiHttpResponse> handleUpdate(
             RestApiHttpRequest req,
             RestApiHttpResponse res,
             ConfigurationPresentation<String> presentation
     ) {
-        try {
-            String updateReq = req
-                    .request()
-                    .content()
-                    .readCharSequence(req.request().content().readableBytes(), UTF_8)
-                    .toString();
-
-            presentation.update(updateReq);
-        } catch (IllegalArgumentException e) {
-            ErrorResult errRes = new ErrorResult("INVALID_CONFIG_FORMAT", e.getMessage());
-
-            res.status(BAD_REQUEST);
-            res.json(Map.of("error", errRes));
-        } catch (ConfigurationValidationException e) {
-            ErrorResult errRes = new ErrorResult("VALIDATION_EXCEPTION", e.getMessage());
-
-            res.status(BAD_REQUEST);
-            res.json(Map.of("error", errRes));
-        } catch (IgniteException e) {
-            ErrorResult errRes = new ErrorResult("APPLICATION_EXCEPTION", e.getMessage());
-
-            res.status(BAD_REQUEST);
-            res.json(Map.of("error", errRes));
-        }
+        String updateReq = req.request().content().toString(StandardCharsets.UTF_8);
+
+        return presentation.update(updateReq)
+                .thenApply(v -> res)
+                .exceptionally(e -> {
+                    if (e instanceof CompletionException) {
+                        e = e.getCause();
+                    }
+
+                    ErrorResult errRes;
+
+                    if (e instanceof IllegalArgumentException) {
+                        errRes = new ErrorResult("INVALID_CONFIG_FORMAT", e.getMessage());
+                    } else if (e instanceof ConfigurationValidationException) {
+                        errRes = new ErrorResult("VALIDATION_EXCEPTION", e.getMessage());
+                    } else if (e instanceof IgniteException) {
+                        errRes = new ErrorResult("APPLICATION_EXCEPTION", e.getMessage());
+                    } else {
+                        throw new CompletionException(e);
+                    }
+
+                    res.status(BAD_REQUEST);
```

Comment:
`BAD_REQUEST` is for the cases when input is invalid. `ConfigurationValidationException` and `IllegalArgumentException` seem to fit this, but what if a handler throws an exception unrelated to validation? In such case `BAD_REQUEST` does not seem suitable.

I suggest to choose the status based on the exception class: if it's related to validation, then we return 400, otherwise 500. Later the assortment of the possible codes could be expanded.

Reply:
I agree, but it is not related to my changes and I would like to avoid adding more code to this PR. Maybe we can incorporate this change in the upcoming refactoring.